pkg/flowaggregator/flowaggregator.go

```diff
@@ -275,10 +275,27 @@ func (fa *flowAggregator) InitAggregationProcess() error {
 }
 
 func (fa *flowAggregator) Run(stopCh <-chan struct{}) {
-	go fa.collectingProcess.Start()
-	defer fa.collectingProcess.Stop()
-	go fa.aggregationProcess.Start()
-	defer fa.aggregationProcess.Stop()
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+	go func() {
+		// Waiting for this function to return on stop makes it easier to set expectations
+		// when testing. Without this, there is no guarantee that
+		// fa.collectingProcess.Start() was called by the time Run() returns.
+		// It also makes more sense to ensure that fa.collectingProcess.Stop() is always
+		// called after fa.collectingProcess.Start().
```

Comment:
This is not ensured, I think it should be removed?

Reply:
yes, thanks for the reminder